app/services/decisions/client_decision_tree.rb

```diff
@@ -55,6 +55,16 @@ def after_has_partner
       end
     end
 
+    def after_client_details
+      if DateStamper.new(form_object.crime_application, 'no_case_type').call
+        edit(:date_stamp)
+      elsif form_object.crime_application.not_means_tested?
```

Comment:
If it already has a datestamp but is not means tested, it is routed differently (if no datestamp yet, it would go to datestamp page and routing onward from there is the same as here)